RecoHI/HiJetAlgos/plugins/HiFJRhoFlowModulationProducer.cc

```diff
@@ -0,0 +1,251 @@
+      eventPlane2Cos += std::cos(2 * pfCandidate.phi());
+      eventPlane2Sin += std::sin(2 * pfCandidate.phi());
+
+      eventPlane3Cos += std::cos(3 * pfCandidate.phi());
+      eventPlane3Sin += std::sin(3 * pfCandidate.phi());
+    }
+  }
+
+  if (!doEvtPlane_) {
+    eventPlane2 = std::atan2(eventPlane2Sin, eventPlane2Cos) / 2.;
+    eventPlane3 = std::atan2(eventPlane3Sin, eventPlane3Cos) / 3.;
+  } else {
+    eventPlane2 = hiEvtPlane[8];
+    eventPlane3 = hiEvtPlane[15];
+  }
+
+  if (nFill >= 100 && eventPlane2 > -99) {
+    const int nPhiBins = std::max(10, nFill / 30);
+
+    std::string name = "phiTestIEta4_" + std::to_string(iEvent.id().event()) + "_h";
+    std::string nameFlat = "phiTestIEta4_Flat_" + std::to_string(iEvent.id().event()) + "_h";
+
+    TH1F* phi_h = new TH1F(name.data(), "", nPhiBins, -TMath::Pi(), TMath::Pi());
+
+    for (auto const& pfCandidate : *pfCands) {
+      if (pfCandidate.eta() < -1.0)
+        continue;
+      if (pfCandidate.eta() > 1.0)
+        continue;
+      if (pfCandidate.pt() < .3)
+        continue;
+      if (pfCandidate.pt() > 3.)
+        continue;
+      if (pfCandidate.particleId() != 1)
+        continue;
+
+      if (doJettyExclusion_) {
+        bool isGood = true;
+        for (auto const& jet : *jets) {
+          if (deltaR2(jet, pfCandidate) < .16) {
+            isGood = false;
+            break;
+          }
+        }
+        if (!isGood)
+          continue;
+      }
+
+      phi_h->Fill(pfCandidate.phi());
+    }
+
+    std::string flowFitForm = "[0]*(1.+2.*([1]*TMath::Cos(2.*(x-[2]))+[3]*TMath::Cos(3.*(x-[4]))))";
+
+    TF1* flowFit_p = new TF1("flowFit", flowFitForm.c_str(), -TMath::Pi(), TMath::Pi());
+    flowFit_p->SetParameter(0, 10);
+    flowFit_p->SetParameter(1, 0);
+    flowFit_p->SetParameter(2, eventPlane2);
+    flowFit_p->SetParameter(3, 0);
+    flowFit_p->SetParameter(4, eventPlane3);
+
+    if (!doFreePlaneFit_) {
+      flowFit_p->FixParameter(2, eventPlane2);
+      flowFit_p->FixParameter(4, eventPlane3);
+    }
+
+    TF1* lineFit_p = new TF1("lineFit", "[0]", -TMath::Pi(), TMath::Pi());
+    lineFit_p->SetParameter(0, 10);
+
+    phi_h->Fit(flowFit_p, "Q", "", -TMath::Pi(), TMath::Pi());
+    phi_h->Fit(lineFit_p, "Q", "", -TMath::Pi(), TMath::Pi());
+
+    rhoFlowFitParamsOut->at(0) = flowFit_p->GetParameter(0);
+    rhoFlowFitParamsOut->at(1) = flowFit_p->GetParameter(1);
+    rhoFlowFitParamsOut->at(2) = flowFit_p->GetParameter(2);
+    rhoFlowFitParamsOut->at(3) = flowFit_p->GetParameter(3);
+    rhoFlowFitParamsOut->at(4) = flowFit_p->GetParameter(4);
+
+    rhoFlowFitParamsOut->at(5) = flowFit_p->GetChisquare();
+    rhoFlowFitParamsOut->at(6) = flowFit_p->GetNDF();
+
+    rhoFlowFitParamsOut->at(7) = lineFit_p->GetChisquare();
+    rhoFlowFitParamsOut->at(8) = lineFit_p->GetNDF();
+
+    delete phi_h;
+
+    delete flowFit_p;
+    delete lineFit_p;
+  }
+
+  iEvent.put(std::move(rhoFlowFitParamsOut), "rhoFlowFitParams");
```

Comment:
The instance name needs to be removed also here
```suggestion
  iEvent.put(std::move(rhoFlowFitParamsOut));
```